apps/dashboard/src/components/workflow-editor/add-step-menu.tsx

```diff
@@ -41,7 +43,7 @@ const MenuItem = ({
 
   return (
     <span
-      onClick={onClick}
+      onClick={!disabled ? onClick : undefined}
```

Comment:
A more neat approch could be defining `const noop = () => {}` at the top of the file outside of the component. Then: 

```suggestion
      onClick={disabled ? noop : onClick }
```